pyro/contrib/minipyro.py

```diff
@@ -201,6 +197,13 @@ def sample(name, fn, *args, **kwargs):
     return msg["value"]
 
 
+# sample is an effectful version of Distribution.sample(...)
+# When any effect handlers are active, it constructs an initial message and calls apply_stack.
+def sample(name, fn, *args, **kwargs):
+    obs = kwargs.pop('obs', None)
+    return start_stack_call_or_apply("sample", name, fn, obs, *args, **kwargs)
+
+
```

Comment:
While code deduplication is nice in the main Pyro library, I feel that in minipyro it is advantageous to explicitly spell out exactly what `sample` and `param` do, with minimal abstraction. One of the original motivations of minipyro was to be less abstract because original-Pyro's abstractions made it difficult to read (even for Pyro devs 🙂).